src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Output.Models.Shared;
+using System;
+using System.Linq;
+using AutoRest.CSharp.Mgmt.Decorator;
+using System.Diagnostics.CodeAnalysis;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    internal class SignatureType
```

Comment:
The name of this type is kind of hard to understand, could we add some xml doc to explain the purpose?
Maybe we need to come up with a new name for this type

Reply:
Added documentation